tests/replace-add-needed.sh

```diff
@@ -0,0 +1,34 @@
+#! /bin/sh -e
+SCRATCH=scratch/$(basename $0 .sh)
+PATCHELF=$(readlink -f "../src/patchelf")
+
+rm -rf ${SCRATCH}
+mkdir -p ${SCRATCH}
+
+cp simple ${SCRATCH}/
+cp libfoo.so ${SCRATCH}/
+cp libbar.so ${SCRATCH}/
+
+
+pushd ${SCRATCH}
+
+libcldd=$(ldd ./simple | grep -oP "(?<=libc.so.6 => )[^ ]+")
+
+# We have to set the soname on these libraries
+${PATCHELF} --set-soname libbar.so ./libbar.so
```

Comment:
Interestingly discussed with @trws that if the shared object doesn't have a _DT_SONAME_, it won't make use of the previously discovered loaded shared object. 

This was discovered while writing this test. You can see below that `libbar.so` was not found, even though its set with an absolute path.
```
ldd ~/code/github.com/NixOS/patchelf/tests/scratch/replace-add-needed/simple
	linux-vdso.so.1 (0x00007ffc90565000)
	/home/fmzakari/code/github.com/NixOS/patchelf/tests/scratch/replace-add-needed/libbar.so (0x00007f532a4c4000)
	/home/fmzakari/code/github.com/NixOS/patchelf/tests/scratch/replace-add-needed/libfoo.so (0x00007f532a4bf000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f532a2da000)
	/nix/store/jsp3h3wpzc842j0rz61m5ly71ak6qgdn-glibc-2.32-54/lib/ld-linux-x86-64.so.2 => /lib64/ld-linux-x86-64.so.2 (0x00007f532a4cb000)
	libbar.so => not found
```